parser.y

```diff
@@ -57,15 +57,30 @@ extern std::unique_ptr<AstNode> program;
 %nterm <std::vector<std::unique_ptr<DeclNode>>> decls
 %nterm <std::unique_ptr<StmtNode>> stmt
 %nterm <std::vector<std::unique_ptr<StmtNode>>> stmts
-%nterm <std::unique_ptr<BlockStmtNode>> block
-%nterm <std::unique_ptr<BlockStmtNode>> main_func
+%nterm <std::unique_ptr<StmtNode>> main_func
 
 %left '='
 %left EQ NE
 %left '<' '>' LE GE
 %left '+' '-'
 %left '*' '/' '%'
 
+// Example: IF '(' expr ')' IF '(' expr ')' stmt • ELSE stmt
```

Comment:
Upon initial inspection, it is unclear what this example is attempting to demonstrate.
I recommend adding the following imperative sentence for clarity:
```suggestion
// Resolve the ambiguity in the "dangling-else" grammar.
// Example: IF '(' expr ')' IF '(' expr ')' stmt • ELSE stmt
```